src/index.js

```diff
@@ -1,15 +1,26 @@
 const arrify = require('arrify');
 const merge = require('deepmerge');
+const path = require('path');
 
 module.exports = ({ config }, options) => {
+  const NODE_MODULES = path.join(__dirname, 'node_modules');
+  const LOADER_EXTENSIONS = /\.vue$/;
+  const styleRule = config.module.rules.get('style');
+  const stylePostCssLoader = styleRule && styleRule.uses.get('postcss');
+  const lintRule = config.module.rules.get('lint');
+
   config.module.rule('vue')
-    .test(/\.vue$/)
+    .test(LOADER_EXTENSIONS)
     .use('vue')
-    .loader(require.resolve('vue-loader'))
-    .options(options);
+      .loader(require.resolve('vue-loader'))
+      .options(options);
+
+  config.resolve.extensions.add('.vue');
```

Comment:
Personally I always write the extension explicitly. After this change you have a freedom to choose between writing the extension and not. This behavior is completely compatible with previous one. If we expect this preset to be used with wide variety of developers we should not limit them in conventions.